api/v1alpha1/contour_types.go

```diff
@@ -345,6 +376,73 @@ const (
 	AWSNetworkLoadBalancer AWSLoadBalancerType = "NLB"
 )
 
+type AzureLoadBalancerParameters struct {
+	// Address is the desired load balancer IP address. If scope is "Internal", address
+	// must reside in same virtual network as AKS and must not already be assigned
+	// to a resource. If address does not reside in same subnet as AKS, the subnet
+	// parameter is also required.
+	//
+	// Address must already exist (e.g. `az network public-ip create`).
+	//
+	// See:
+	// 	 https://docs.microsoft.com/en-us/azure/aks/static-ip#create-a-service-using-the-static-ip-address
+	// 	 https://docs.microsoft.com/en-us/azure/aks/internal-lb#specify-an-ip-address
+	//
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=253
+	// +optional
+	Address *string `json:"address,omitempty"`
+
+	// ResourceGroup is the resource group name where the "address" resides. Relevant
+	// only if scope is "External".
+	//
+	// Omit if desired IP is created in same resource group as AKS cluster.
+	//
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:valodation:MaxLength=90
```

Comment:
s/valodation/validation/

It's surprising that `controller-gen` does not surface an error with this misspelling.